keytree/src/lib.rs

```diff
@@ -25,7 +31,19 @@ impl Xprv {
         let scalar = Scalar::random(&mut rng);
         let mut dk = [0u8; 32];
         rng.fill_bytes(&mut dk);
-        Xprv { scalar, dk }
+
+        let mut transcript = Transcript::new(b"Keytree.intermediate");
```

Comment:
Since it is going to be used for both intermediate nodes and leafs, i’d change label to “.derivation”, and adjust the spec accordingly (see also my comments on the domain separation in another PR).

Reply:
Right, thanks.